tests/conftest.py

```diff
@@ -0,0 +1,350 @@
+
+        if config.getoption("--4C") and "fourc" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--ArborX") and "arborx" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--CubitPy") and "cubitpy" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--Performance") and "performance" in markers:
+            selected_tests.append(item)
+
+        if not markers and not config.getoption("--exclude-standard-tests"):
+            selected_tests.append(item)
+
+    deselected_tests = list(set(items) - set(selected_tests))
+
+    items[:] = selected_tests
+    config.hook.pytest_deselected(items=deselected_tests)
+
+
+@pytest.fixture(scope="session")
+def reference_file_directory() -> Path:
+    """Provide the path to the reference file directory.
+
+    Returns:
+        Path: A Path object representing the full path to the reference file directory.
+    """
+
+    testing_path = os.path.abspath(os.path.dirname(__file__))
+    return Path(testing_path) / "reference-files"
+
+
+@pytest.fixture(scope="function")
+def current_test_name(request: pytest.FixtureRequest) -> str:
+    """Return the name of the current pytest test.
+
+    Args:
+        request: The pytest request object.
+
+    Returns:
+        str: The name of the current pytest test.
+    """
+
+    return request.node.name
+
+
+@pytest.fixture(scope="function")
+def get_string() -> Callable:
+    """Return function to get string from file.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _get_string(path: Path) -> str:
+        """Get string from file.
+
+        Args:
+            path: Path to file.
+
+        Returns:
+            String of file
+        """
+
+        if not path.is_file():
+            raise FileNotFoundError(f"File {path} does not exist!")
+
+        with open(path, "r") as file:
+            return file.read()
+
+    return _get_string
+
+
+@pytest.fixture(scope="function")
+def compare_results() -> Callable:
+    """Return function to compare either string or files.
+
+    Necessary to enable the function call through pytest fixtures.
+    """
+
+    def _compare(
+        reference: Union[Path, str],
+        result: Union[Path, str],
+        rtol: float = 1e-14,
+        atol: float = 1e-14,
+        **kwargs,
+    ) -> bool:
+        """Comparison between reference and result with relative or absolute
+        tolerance.
+
+        Args:
+            reference: The reference string or path to the reference file.
+            result: The result string or path to the result file.
+            rtol: The relative tolerance.
+            atol: The absolute tolerance.
+
+        Returns:
+            bool: true if comparison is successful, false otherwise
+        """
+
+        if type(reference) is not type(result):
+            raise TypeError("Reference and result must be of the same type.")
```

Comment:
We have a lot of tests, where we pass a string that is created during testing and compare it to a string saved in a reference file. This would mean that we have to first manually read the file and save it to a string variable and pass that to the function. That is likely just one or two lines of code, but they will be repeated a lot. Why not allow file vs. string comparisons?